build/utils/dotnet/dependencies/packagesconfig.go

```diff
@@ -315,10 +316,21 @@ type group struct {
 func xmlUnmarshal(content []byte, obj interface{}) (err error) {
 	err = xml.Unmarshal(content, obj)
 	if err != nil {
-		// Sometimes the nuspec file is wrongly encoded in utf-16, the actual encoding is utf-8 but the xml header has 'encoding="utf-16"' key.
+		// Failed while trying to parse xml file. Nuspec file could be an utf-16 encoded file.
```

Comment:
Yes! This is the issue I was running into. Now I see why the original fix didn't work.